packages/services/src/interfaces/storageService.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Extensible
+} from "@/types";
+
+import {
+  SignedVerifiableCredential
+} from "./verifiableCredentialService";
+
+/**
+ * Result of storing a verifiable credential in external storage
+ */
+export type StorageResponse = {
+  uri: string;
+  hash: string;
+} & Extensible;
+
+/**
+ * Configuration for storing a credential in an external storage service
+ */
+export type StorageConfig = {
+  url: string;
+  params: { 
+    bucket?: string;
+  } & Extensible;
+  options: {
+    method: 'POST' | 'PUT';
+    headers: Record<string, string>;
+  } & Extensible;
+} & Extensible;
+
+/**
+ * Service responsible for persisting verifiable credentials
+ */
+export interface IStorageService {
+  /**
+   * Stores a signed verifiable credential using the provided storage configuration
+   */
+  store(
+    config: StorageConfig, 
+    crendential: SignedVerifiableCredential
+  ): Promise<StorageResponse>
+}
```

Comment:
This method would require some kind of token to be able to store the credential, wouldn't it? I don't see where that is (can't see it in the `StorageConfig`?

Reply:
Sorry, what I meant was that this actual request function would need to include the token - ah, or you mean that the implementation will access the token directly (from the env or app config) - ok, I hadn't realised it has access to the implementation has access to the app config without the data being passed in. Great.